File: app/src/main/java/io/github/howiezuo/unsplash/AppApplication.java
package io.github.howiezuo.unsplash;

import android.app.Application;

import java.io.IOException;
import java.util.Properties;


public class AppApplication extends Application {

    private static AppApplication INSTANCE;

    private String clientId;
    private String clientSecret;

    @Override
    public void onCreate() {
        super.onCreate();

        INSTANCE = this;

        Properties properties = new Properties();
        try {
            properties.load(getAssets().open("private.properties"));
            clientId = properties.getProperty("client.id");
            clientSecret = properties.getProperty("client.secret");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static AppApplication getInstance() {
        return INSTANCE;
    }

    public String getClientId() {
        return clientId;
    }

    public String getClientSecret() {
        return clientSecret;
    }
}

File: app/src/main/java/io/github/howiezuo/unsplash/app/AppApplication.kt
package io.github.howiezuo.unsplash.app

import android.app.Application
import java.io.IOException
import java.util.*


class AppApplication : Application() {

    var clientId: String? = null
        private set
    var clientSecret: String? = null
        private set

    override fun onCreate() {
        super.onCreate()

        instance = this

        val properties = Properties()
        try {
            properties.load(assets.open("private.properties"))
            clientId = properties.getProperty("client.id")
            clientSecret = properties.getProperty("client.secret")
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    companion object {
        lateinit var instance: AppApplication
            private set
    }

}
